api/client/storagejobs.go

```diff
@@ -14,6 +14,48 @@ type StorageJobs struct {
 	client userPb.UserServiceClient
 }
 
+// ListSelect specifies which StorageJobs to list.
+type ListSelect int32
+
+const (
+	// All lists all StorageJobs and is the default.
+	All ListSelect = iota
+	// Queued lists queued StorageJobs.
+	Queued
+	// Executing lists executing StorageJobs.
+	Executing
+	// Final lists final StorageJobs.
+	Final
+)
+
+// ListConfig controls the behavior for listing StorageJobs.
+type ListConfig struct {
+	// CidFilter filters StorageJobs list to the specified cid. Defaults to no filter.
+	CidFilter string
+	// Limit limits the number of StorageJobs returned. Defaults to no limit.
+	Limit uint64
+	// Ascending returns the StorageJobs ascending by time. Defaults to false, descending.
+	Ascending bool
+	// Select specifies to return StorageJobs in the specified state.
+	Select ListSelect
+	// NextPageToken sets the slug from which to start building the next page of results.
+	NextPageToken string
+}
+
+type summaryConfig struct {
+	cid string
+}
+
+// SummaryOption configures a storageJobsConfig.
+type SummaryOption = func(*summaryConfig)
+
+// WithCid filters the results to the specified data cid.
+func WithCid(cid string) SummaryOption {
+	return func(conf *summaryConfig) {
+		conf.cid = cid
+	}
+}
+
```

Comment:
They are already different... the admin client supports filtering by user id and cid, while the user client only supports cid. I agree I don't like the duplicated parts. Will think about the best way to share code here.

Reply:
Yeah, not sure though. If only simplifies things. If you feel covering both cases is uglier, doesn't seem to worth it. (Since some fields doesn't make sense in other case).